monai/inferers/inferer.py

```diff
@@ -244,11 +264,27 @@ def _aggregate(self, outputs, locations, batch_size, mergers, ratios):
                 merger.aggregate(out_patch, out_loc)
 
     def _get_output_shape(self, inputs, out_patch, ratio):
-        """Define the shape of output merged tensors"""
-        in_spatial_shape = inputs.shape[2:]
-        out_spatial_shape = tuple(round(s * r) for s, r in zip(in_spatial_shape, ratio))
-        output_shape = out_patch.shape[:2] + out_spatial_shape
-        return output_shape
+        """Define the shape of output merged tensors (non-padded and padded)"""
+        if (
+            self.splitter is None
+            or not hasattr(self.splitter, "get_input_shape")
+            or not hasattr(self.splitter, "get_output_shape")
+        ):
+            return None, None
+
+        # input spatial shapes
+        original_spatial_shape = self.splitter.get_input_shape(inputs)
+        padded_spatial_shape = self.splitter.get_output_shape(inputs)
+
+        # output spatial shapes
+        output_spatial_shape = tuple(round(s * r) for s, r in zip(original_spatial_shape, ratio))
+        padded_output_spatial_shape = tuple(round(s * r) for s, r in zip(padded_spatial_shape, ratio))
+
+        # output shapes
+        crop_shape = out_patch.shape[:2] + output_spatial_shape
+        output_shape = out_patch.shape[:2] + padded_output_spatial_shape
```

Comment:
I feel confusing about the `crop_shape` and `output_shape` here, could you please help help explain them? Maybe `self.crop=True` should not work with `output_shape` arg together?

Thanks.

Reply:
Sure, `output_shape` is the shape of the output tensor (might be padded) that needs to be created to store the merged output results and it is always required to properly initialize the merger. However, `crop_shape` is an optional shape to which the output tensor is cropped to have the same shape as the input (non-padded). Thus is if there is no padding involved, both are the same.

Reply:
I changed the var names and some changes to the classes to make it less confusing. Now, merger has two shapes: `merging_shape` and `final_shape`.
1. `merging_shape` is the shape of the tensor required to aggregate and merge patches. For instance, it might need to be padded to include the patches crossing the boundaries.
2. `final_shape` is the shape of the final merged output. By default it is the same as `merging_output` but if user provide it, the output tensor will be cropped to have `final_shape` spatial shape.
3. Per previous comment, `crop` is renamed to `match_spatial_shape` and it is a feature of `PatchInferer` and not `Merger`, which tells the `PatchInferer` to find the original input shape of the input data and provide it to the Merger to crop the output to that shape. Thus, I believe it should remain as a main argument of `PatchInferer` and not `meger_kwargs`

Does it make sense to you now? Thanks

Reply:
Thanks for the quick update and detailed explanation.
"final_shape" may be a little confusing because it's hard to "guess" which step is "final" here.
And not very sure the `shape` means input or output.
How about change `merging_shape` and `final_shape` to `merge_input_shape` and `merge_output_shape`?
What do you think?

Thanks.

Reply:
@Nic-Ma I agree with you that merging and final are not very clear. On the other hand, `merge_input_shape` is also can be confusing since it is the output shape and the input to the merger is basically patches. Let me rethink it to come up with some alternatives.

Reply:
What about `merged_shape`, which is the shape of the merged tensor, and `crop_to_shape` (or `cropped_shape`), which is the shape that we *optionally* want to crop to?

Reply:
I think in this way, it is clear that we have a "merged" tensor shape and a shape to possibly "crop" the merged shape. How does it feel? Are you good with them?